include/net/net_timeout.h

```diff
@@ -31,8 +31,9 @@ extern "C" {
 /**
  * Symbol to indicate that the caller wants the timeout to be waited forever.
  * This can be used when a network API expects a millisecond timeout.
+ * Use SYS_FOREVER_MS value directly instead.
  */
-#define NET_WAIT_FOREVER (-1)
+#define NET_WAIT_FOREVER SYS_FOREVER_MS
```

Comment:
Do we still need this symbol then?

Reply:
If someone is still using the symbol, their code would not stop working. Perhaps that is not a big issue thou as we just introduced the symbol.

Reply:
After some pondering, I removed the NET_WAIT_FOREVER altogether.